spark/src/main/java/org/apache/iceberg/spark/source/SparkAppenderFactory.java

```diff
@@ -85,6 +72,52 @@
     this.posDeleteRowSchema = posDeleteRowSchema;
   }
 
+  public static SparkAppenderFactoryBuilder builderFor(Table table, Schema writeSchema, StructType dsSchema) {
+    return new SparkAppenderFactoryBuilder(table, writeSchema, dsSchema);
+  }
+
+  public static class SparkAppenderFactoryBuilder {
```

Comment:
Since the class `SparkAppenderFactory`  was not marked as `public`, so I think the newly introduced `SparkAppenderFactoryBuilder` also don't have to be `public` ? 

btw,   I think we could just use the name `Builder` because it's in the class `SparkAppenderFactory`,  the prefix `SparkAppenderFactory` seems to be redundant.

Reply:
Done